catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java

```diff
@@ -145,71 +138,88 @@ public String getId() {
-      boolean hasDerivedImage = hasOriginalDerivedResource(metacard);
-
-      canHandle = isImageNitf && hasLocation && hasDerivedImage;
-    }
-
-    return canHandle;
-  }
-
-  private boolean hasOriginalDerivedResource(Metacard metacard) {
-    Attribute attribute = metacard.getAttribute(Core.DERIVED_RESOURCE_URI);
-
-    return Stream.of(attribute)
-        .filter(Objects::nonNull)
-        .flatMap(a -> a.getValues().stream())
-        .filter(String.class::isInstance)
-        .map(String.class::cast)
-        .anyMatch(this::hasOriginalQualifier);
-  }
-
-  private boolean hasOriginalQualifier(String uriString) {
-    try {
-      URI derivedResourceUri = new URI(uriString);
-
-      // find the #original URI fragment for local or =original param for remote
-      if (ORIGINAL_QUALIFIER.equals(derivedResourceUri.getFragment())
-          || ORIGINAL_QUALIFIER.equals(getQualifierForRemoteResource(uriString))) {
-        return true;
-      }
-    } catch (URISyntaxException use) {
-      LOGGER.debug("Could not parse URI string [{}]", uriString);
+      final Metacard metacard = (Metacard) subject;
+
+      final boolean isImageNitf =
+          NITF_IMAGE_METACARD_TYPE.equals(metacard.getMetacardType().getName());
+      final boolean hasLocation = hasValidLocation(metacard.getLocation());
+      // The chipping transformer requires the NITF resource, the overview derived image
+      // resource, and the original derived image resource.
+      final boolean hasNitfResource = metacard.getResourceURI() != null;
+      boolean hasOriginalDerivedImageResource = getOriginalDerivedResourceUri(metacard).isPresent();
+      // assume that if there is an original, there is also an overview
+      boolean hasOverviewDerivedImageResource = hasOriginalDerivedImageResource;
+
+      return isImageNitf
+          && hasLocation
+          && hasNitfResource
+          && hasOverviewDerivedImageResource
+          && hasOriginalDerivedImageResource;
+    } else {
+      return false;
     }
-    return false;
   }
 
   private boolean hasValidLocation(String location) {
-    boolean hasValidLocation = false;
-
     if (StringUtils.isNotBlank(location)) {
       try {
         // parse the WKT location to determine if it has valid format
-        WKTReader wktReader = new WKTReader(geometryFactory);
+        final WKTReader wktReader = new WKTReader(geometryFactory);
         wktReader.read(location);
-        hasValidLocation = true;
+        return true;
       } catch (ParseException e) {
-        LOGGER.debug("Location [{}] is invalid, cannot chip this image", location);
+        LOGGER.debug("Location [{}] is invalid. Cannot chip this image", location);
+      }
+    }
+
+    return false;
+  }
+
+  private static Optional<URI> getOriginalDerivedResourceUri(final Metacard metacard) {
+    final Attribute derivedResourceUriAttribute =
+        metacard.getAttribute(Metacard.DERIVED_RESOURCE_URI);
+    if (derivedResourceUriAttribute != null) {
```

Comment:
✏️  Also it seems like a few of these ifs could be inverted to reduce the complexity/nesting of the code if you'd like